docs/modules/ROOT/pages/tutorial/02_resource-definition.adoc

```diff
@@ -0,0 +1,105 @@
+include::partial$attributes.adoc[]
+
+= Policy authoring
+
+NOTE: The policies for this section can be found https://github.com/cerbos/tutorial/tree/main/src/03-resource-definition/cerbos[on Github].
+
+== Authentication roles
+
+To begin with Cerbos needs to know about the basic roles which are provided by your authentication provider. In the case of Cerbforce, https://cerbos.dev/ecosystem/cerbos-auth0[Auth0] provides a role of either `ADMIN` or `USER` for all profiles. This is important when starting to define access to resources below - for now just make a note of them.
+
+== Resources
+
+The best place to start with defining xref:policies:index.adoc[policies] is listing out all the resources and their actions that exist in the system. A resource is an entity type that users are authorized to access.
+
+In the case of Cerbforce some of the resources and actions are as follows:
+
+[caption=]
+[%header,cols=".^1,.^1,3m",grid=rows]
+|===
+|Resource | Actions 
+|User | Create, Read, Update, Delete
+|Company | Create, Read, Update, Delete
+|Contact | Create, Read, Update, Delete
+|===
+
+With this as a start, you can begin creating your first Cerbos policy - a xref:policies:resource_policies.adoc[resource policy].
+
+
+== Resource policies
+
+Taking the user resource as an example, the most basic resource policy can be defined like below:
+
+[source,yaml]
+----
+apiVersion: api.cerbos.dev/v1
+resourcePolicy:
+  version: "default" 
+  resource: "user"
+  rules:
+    - actions: 
+        - create
+        - read
+      effect: EFFECT_ALLOW
+      roles:
+        - user 
+
+    - actions: 
+        - create
+        - read
+        - update
+        - delete
+      effect: EFFECT_ALLOW
+      roles:
+        - admin         
+----
+
+The structure of a resource policy requires a name to be set on the `resource` key and then a list of rules is defined. A rule defines a list of actions on the resource, the effect of the rule (`EFFECT_ALLOW` or `EFFECT_DENY`) and then fields to state who this applies to - in this simple case a list of `roles` which is checked for in the roles of the user making the request.
+
+In this case, a request made for a principal with a role of `user` is granted only `create` and `read` actions whilst an `admin` role can also perform `update`, `delete` actions.
+
+The full documentation for resource policies can be found xref:policies:resource_policies.adoc[here].
+
+== Wildcard action
+
+To simplify things further, admins to be able to do every action so a special `*` wildcard action can be used to keep things clean:
+
+[source,yaml]
+----
+{{#include ./cerbos/policies/user.yaml}}  
```

Comment:
Thanks. I think I'm going to have to copy the yaml across direct, given the previous `include` was referencing a file in the local github.

Reply:
I think you can add the files to `examples` directory and then `include` them. We do that in a few places (e.g. https://raw.githubusercontent.com/cerbos/cerbos/main/docs/modules/ROOT/pages/quickstart.adoc)

https://docs.antora.org/antora/latest/page/include-an-example/

Reply:
Happy to do that. My thinking was that we'd be maintaining two file copies either way so pasting in directly just cut out an additional reference layer (assuming resources are only referenced once of course).

I had been wondering about the link to the github repo. Now the content's been merged into our main docs (and particularly if we no longer reference the files), it almost renders the repo defunct. Which is a shame, because it's nicely laid out and clear.

Reply:
Well, Antora supports using multiple git repos for content. So, the book could potentially still be in its own repo and get pulled into the main repo only during the build. 

I haven't tried it myself, so I don't know how easy/hard it is though. https://docs.antora.org/antora/latest/playbook/configure-content-sources/

Reply:
Oh cool! I'll check that out, thanks.

Reply:
Think I'm at risk of going down a rabbit hole on this one - looks like configuring content sources requires explicit `antora.yml` configuration and directory structures which would require some changes to the remote `books` repo, and I question the need for this if we're just consolidating the content from that repo over here anyway. I am however a complete antora noob, so apologies if I'm missing the mark here! Regardless, I'll look at bringing the content over locally.